tests/roles/development_environment/files/pre_launch.bash

```diff
@@ -17,6 +17,44 @@ function wait_for_status() {
     done
 }
 
+function create_volume_resources() {
+    # create a data volume
+    if ! ${BASH_ALIASES[openstack]} volume show disk ; then
+        ${BASH_ALIASES[openstack]} volume create --image cirros --size 1 disk
+        wait_for_status "volume show disk" "test volume 'disk' creation"
+    fi
+
+    # create volume snapshot
+    if ! ${BASH_ALIASES[openstack]} volume snapshot show snapshot ; then
+        ${BASH_ALIASES[openstack]} volume snapshot create --volume disk snapshot
+        wait_for_status "volume snapshot show snapshot" "test volume 'disk' snapshot availability"
+    fi
+
+    # Add volume to the test VM
+    if ${BASH_ALIASES[openstack]} volume show disk -f json | jq -r '.status' | grep -q available ; then
+        ${BASH_ALIASES[openstack]} server add volume test disk
+    fi
+}
+
+function create_backup_resources() {
+    # create volume backup
+    if ! ${BASH_ALIASES[openstack]} volume backup show backup; then
+        ${BASH_ALIASES[openstack]} volume backup create --name backup disk
```

Comment:
${BASH_ALIASES[openstack]} volume backup create --name backup disk --force

but I'm wondering if we really want to create the backup after creating server from volume.
backing up before server create might make this more tedious by adding another function and calling it.
I'm okay with the current workflow just we need the --force flag here.